lint/util.py

```diff
@@ -49,22 +49,13 @@ def get_syntax(view):
 
     or the syntax it is mapped to in the "syntax_map" setting.
     """
-    syntax_re = re.compile(r'(?i)/([^/]+)\.(?:tmLanguage|sublime-syntax)$')
-    view_syntax = view.settings().get('syntax', '')
-    mapped_syntax = ''
-
-    if view_syntax:
-        match = syntax_re.search(view_syntax)
-
-        if match:
-            view_syntax = match.group(1).lower()
-            from .persist import settings
-            mapped_syntax = settings.get(
-                'syntax_map', {}).get(view_syntax, '').lower()
-        else:
-            view_syntax = ''
-
-    return mapped_syntax or view_syntax
+    from .persist import settings
+    syntax_map = settings.get('syntax_map')
```

Comment:
Default value?

Reply:
We have gone great l length to separate data from code, which have been intertwined pre-SL4.

The default settings are now read-only. Also, we are going to implement a settings validation via the new `jsonschema` dependency.

Hence there will always be a syntax-map provided.